msm/src/circuit_design/capabilities.rs

```diff
@@ -45,3 +45,59 @@ where
 {
     fn lookup(&mut self, lookup_id: LT, value: &Self::Variable);
 }
+
+////////////////////////////////////////////////////////////////////////////
+// Helpers
+////////////////////////////////////////////////////////////////////////////
+
+/// Write an array of values simultaneously.
+pub fn read_column_array<F, Env, const ARR_N: usize, CIx: ColumnIndexer, Func>(
```

Comment:
Should it not be in the trait `ColAccessCap`? Same for the functions below (into `ColWriteCap`).

Reply:
Big question. I prefer this style for now because then people outside of our library can define these methods. It seems generally more extendable -- you don't have to modify the trait to add another helper. Maybe they can be in a `capabilities::caphelpers` module?

Reply:
https://stackoverflow.com/questions/58157367/how-can-i-add-extension-methods-to-a-trait-with-associated-types-lying-in-a-diff